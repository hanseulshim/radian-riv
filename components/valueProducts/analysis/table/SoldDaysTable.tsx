import React, { useState, useCallback } from 'react'
import Table from 'components/common/Table'
import { formatPrice } from 'utils'

interface Props {
  tableData: any[]
}

export default function SoldDays({ tableData }: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])

  const fetchData = useCallback(async () => {
    const width = 95
    setColumns([
      {
        Header: '',
        accessor: 'category',
        align: 'right'
      },
      {
        Header: 'SOLD DAYS',
        columns: [
          {
            Header: 'Active',
            accessor: row => formatPrice(row['active']),
            width: 80,
            align: 'right'
          },
          {
            Header: 'Pending',
            accessor: row => formatPrice(row['pending']),
            width: 80,
            align: 'right'
          },
          {
            Header: '0-90',
            accessor: row => formatPrice(row['0-90']),
            width: 80,
            align: 'right'
          },
          {
            Header: '91-180',
            accessor: row => formatPrice(row['91-180']),
            width,
            align: 'right'
          },
          {
            Header: '180-270',
            accessor: row => formatPrice(row['180-270']),
            width,
            align: 'right'
          },
          {
            Header: '271-365',
            accessor: row => formatPrice(row['271-365']),
            width,
            align: 'right'
          }
        ]
      }
    ])
    setData(tableData)
  }, [tableData])

  return (
    <Table columns={columns} data={data} fetchData={fetchData} width={700} />
  )
}
