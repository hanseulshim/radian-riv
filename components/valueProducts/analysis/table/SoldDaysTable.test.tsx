import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import SoldDaysTable from './SoldDaysTable'

describe('SoldDaysTable', () => {
  let component
  beforeAll(() => {
    const props = {
      tableData: []
    }
    component = shallow(<SoldDaysTable {...props} />)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
