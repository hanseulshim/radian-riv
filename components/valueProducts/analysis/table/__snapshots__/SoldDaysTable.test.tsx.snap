// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoldDaysTable should match snapshot 1`] = `
<Table
  columns={Array []}
  data={Array []}
  fetchData={[Function]}
  width={700}
/>
`;
